### DISCUSSION TITLE
posts/Go/import_local_package_in_go_module/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#340] [Announcements]
https://www.liwenzhou.com/posts/Go/import_local_package_in_go_module/ 

 李文周的Blog Go语言教程 golang中文教程 李文周博客 七米 qimi go.mod module 依赖管理 导入 包 package 1.14

**Comment by yiliyili** [comment]
packagedemo是笔误,需改为moduledemo

**Comment by Q1mi** [comment]
> @yiliyili
> packagedemo是笔误,需改为moduledemo

感谢反馈，已更正。

**Comment by tyf740083371** [comment]
第一次使用go moulde，应该设置一下go环境配置GO111MODULE；set GO111MODULE=auto

**Comment by XcodeRole** [comment]
> @tyf740083371
> 第一次使用go moulde，应该设置一下go环境配置GO111MODULE；set GO111MODULE=auto

应该不用吧，默认就是auto

**Comment by xyf1096415969** [comment]
replace "mypackage" => "../mypackage"请问这句话的左边的"mypackage"代表什么意思？是目录名？还是包名
?

**Comment by Q1mi** [comment]
> @xyf1096415969
> replace "mypackage" => "../mypackage"请问这句话的左边的"mypackage"代表什么意思？是目录名？还是包名
> ?

包名

**Comment by jamietu** [comment]
我真的卡在這種問題好幾天, 找了好幾篇文章都沒有您這篇的詳細, 真的非常感謝.

**Comment by harry-fan** [comment]
GO111MODULE=on，i get error: package xxx is not in GOROOT
i find many func, not fix it.

**Comment by Q1mi** [comment]
> @harry-fan
> GO111MODULE=on，i get error: package xxx is not in GOROOT
> i find many func, not fix it.

if u write code in 'go module' mode, then u should not write code under GOPATH. move to another dir out GOPATH

**Comment by liuluqqzj** [comment]
浅显易懂，好文

**Comment by seabra2** [comment]
现在已经直接在根目录下创建一个 go.work 文件就行了
